### ISSUE TITLE
aws_kinesis_firehose_delivery_stream splunk configuration incompatible with extended_s3_configuration 

### ISSUE THREAD

**Issue by demaistos** [hashicorp/terraform-provider-aws#22091]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* aws_kinesis_firehose_delivery_stream

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
resource "aws_kinesis_firehose_delivery_stream" "extended_s3_stream" {
  name        = "terraform-kinesis-firehose-extended-s3-test-stream"
  destination = "extended_s3"

  extended_s3_configuration {
    role_arn   = aws_iam_role.firehose_role.arn
    bucket_arn = aws_s3_bucket.bucket.arn
  }

  splunk_configuration {
    hec_endpoint               = "https://http-inputs-mydomain.splunkcloud.com:443"
    hec_token                  = "51D4DA16-C61B-4F5F-8EC7-ED4301342A4A"
    hec_acknowledgment_timeout = 600
    hec_endpoint_type          = "Event"
    s3_backup_mode             = "FailedEventsOnly"
  }
}

```

### Debug Output

```hcl
splunk_configuration need s3_configuration 
```

### Expected Behavior

I would like to use the need block extended_s3_configuration because i need to custom the error prefix bug but there is a problème with my block splunk_configuration . The splunk_configuration want the deprecated block s3_configuration instead extended_s3_configuration . I think splunk_configuration must support the new block extended_s3_configuration.

### Actual Behavior

I can't deploy my stack because there is a incompatiblity with splunk_configuration and extended_s3_configuration

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

**Comment by ewbankkit** [comment]
Relates #22309.

**Comment by johnsonaj** [comment]
Closed in #31138. Merged to `main` in #31392
